elasticache/common.go

```diff
@@ -11,15 +12,19 @@ import (
 )
 
 // ResourceID returns an ID that can be used to uniquely identify resources built with the given prefix
-func ResourceID(prefix string) string {
-	id := os.Getenv("ACORN_EXTERNAL_ID")
-	id = strings.ReplaceAll(id, ".", "")
-
-	if len(id) > 40 {
-		return id[:40]
-	} else {
-		return id
+func ResourceID(clusterName string, prefix string) *string {
+	externalIdHash := md5.Sum([]byte(os.Getenv("ACORN_EXTERNAL_ID")))
+	clusterName = clusterName + "-" + hex.EncodeToString(externalIdHash[:])
```

Comment:
This will always generate the same output given the same input?

Reply:
I believe so. I'll add a test to confirm.

Reply:
Test confirms it does. There's no randomness/salt.